### DISCUSSION TITLE
Help With Completion

### DISCUSSION THREAD

**Discussion by kyp0717** [nix-community/kickstart-nix.nvim#50] [Q&A] [answered]
Hello.
I am unable to figure out how to navigate and choose the correct choice in the completion pop-up window.
I try `tab`, `ctrl-f` and others key-combo but none of it worked.
I try to look in the `completion.lua` and `keymap.lua` to see how completion was configured but I could not find the code for completion keymap.

Please advise.
Thank you.

**Comment by mrcjkb** [comment, marked as answer]
Hey :wave: 

The keymaps defined in `completion.lua` are chosen to be close to what you get with native Neovim:
`<C-n>` and `<C-p>` to navigate and `<C-y>` (for "yes") to select a completion.

nvim-cmp provides a `mapping` DSL via its `setup` function to configure this:

https://github.com/nix-community/kickstart-nix.nvim/blob/f6fd1fa5555a5d9b092a328dfd4212aa2739aeba/nvim/plugin/completion.lua#L55-L103

I advise against using `<TAB>`, as this will make it harder for you to adapt if you are using vim on a remote server.

**Reply by kyp0717** [reply]
Thanks for your prompt respond!
